android-lib/src/main/java/com/kinvey/android/cache/ClassHash.java

```diff
@@ -380,7 +380,7 @@ private static RealmObjectSchema createSchemeFromClass(String name, DynamicRealm
             if (fieldInfo == null){
                 continue;
             }
-            if (fieldInfo.getType().isArray() || Collection.class.isAssignableFrom(fieldInfo.getType())){
```

Comment:
@yuliya-guseva Can you explain more why this condition was causing an exception?

Reply:
This class works with Realm tables and contains a complicated logic.
This part of the condition was logically incorrect because we go into this part of the code only if the field is an array and the type of the array inherits `GenericJson`. In this case, the `RealmObjectSchema.addRealmListField` method is called and the field is created in the table in the form of ArrayList. But for the array this method should not be called.